vernac/vernacentries.ml

```diff
@@ -59,17 +59,21 @@ module DefAttributes = struct
     canonical_instance : bool;
     typing_flags : Declarations.typing_flags option;
     using : Vernacexpr.section_subset_expr option;
+    nonuniform : bool
   }
 
-  let parse f =
+  let parse ?(coercion=false) f =
     let open Attributes in
-    let (((((locality, deprecated), polymorphic), program), canonical_instance), typing_flags), using =
-      parse Notations.(locality ++ deprecation ++ polymorphic ++ program ++ canonical_instance ++ typing_flags ++ using) f
+    let ((((((locality, deprecated), polymorphic), program), canonical_instance), typing_flags), using), nonuniform =
+      parse Notations.(locality ++ deprecation ++ polymorphic ++ program ++ canonical_instance ++ typing_flags ++ using ++ ComCoercion.nonuniform) f
```

Comment:
```suggestion
    let nonuniform = if coercion then ComCoercion.nonuniform else Notations.return None in
    let ((((((locality, deprecated), polymorphic), program), canonical_instance), typing_flags), using), nonuniform =
      parse Notations.(locality ++ deprecation ++ polymorphic ++ program ++ canonical_instance ++ typing_flags ++ using ++ nonuniform) f
```
and remove the `if not coercion` check below

Reply:
Nice, thanks!